#BN-23:Buyer should be able to clear their cart

uwdjany opened the pull request:
# PR Description
https://trello.com/c/KWfchsBs/23-buyer-should-be-able-to-clear-their-cart
List all the tasks and sub-tasks that were assigned to you Make sure you check every completed task. list them in this style
- [x] The buyer's cart should be emptied
- [x] The cart total should be reset to zero
- [x] A cart cleared confirmation message should be sent to the frontend to be displayed to the buyer
# Description of tasks that were expected to be completed and that are not yet completed
- [x] None
# How this may be tested?
Please describe the tests that you ran to verify your functionalities. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.
1. run `npm i` to install all required dependencies
2. run `npm run dev`
3. step 3
4. step 4...
# Screenshots (If appropriate/applicable)
Provide screenshots related to your PR in case it's appropriate and applicable
# Please check this Checklist before you submit your PR:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My code generates no warnings
- [x] There are no vulnerabilities

kunda4 commented on src/controllers/product/clearCartController.js:
```diff
@@ -0,0 +1,45 @@
+
+const clearCart = async (req, res) => {
+  try {
+    // const buyer = req.user?.id; // using optional chaining operator to avoid undefined error
+    // if (!buyer) {
+    //   return res.json({
+    //     status: 400,
+    //     message: 'User not found',
+    //   });
+    // }
```
Remove this commented Line

kunda4 commented on src/database/models/cart.js:
```diff
@@ -15,8 +15,8 @@ module.exports = (sequelize, DataTypes) => {
       product_id: DataTypes.STRING,
-      quantity: DataTypes.INTEGER,
-      date_added: DataTypes.STRING,
+      // quantity: DataTypes.INTEGER,
+      //date_added: DataTypes.STRING,
```
Remove this too

uwdjany replied:
Done!

kunda4 reviewed (commented):
Well done djanati just for making code look great remove that commented line

uwdjany committed:
feat(cart)  Buyer should be able to clear their cart
 -Buyer should be able to reset their cart to zero
-Buyer should be able to empty their cart

[Delivers #cp-23]

uwdjany committed:
feat(cart)  Buyer should be able to clear their cart

 -Buyer should be able to reset their cart to zero
-Buyer should be able to empty their cart

[Delivers #cp-23]

dushimeemma reviewed (dismissed):
Your commit message should match standard.
example on [wiki](https://github.com/atlp-rwanda/engineering-playbook/wiki)

dushimeemma reviewed (changes requested):
Your commit message should match standard.
example on [wiki](https://github.com/atlp-rwanda/engineering-playbook/wiki)
